crates/forge-runner/src/gas.rs

```diff
@@ -138,6 +143,30 @@ fn get_state_resources(
     })
 }
 
+fn reduce_state_resources_after_strk_token_predeployment(
+    state_resources: &mut StateResources,
+) -> StateResources {
+    // STRK predeployment results in state changes. To avoid including them in gas cost
+    // we need to reduce the state changes count. These calculations could be slightly inaccurate only if
+    // someone would modify storage cells which are changed in the STRK constructor.
+
+    state_resources
+        .state_changes_for_fee
+        .state_changes_count
+        .n_storage_updates -= 10;
+    state_resources
+        .state_changes_for_fee
+        .state_changes_count
+        .n_class_hash_updates -= 1;
+    state_resources
+        .state_changes_for_fee
+        .state_changes_count
+        .n_modified_contracts -= 1;
```

Comment:
In case someone interacts with the token contract at all, it won't be reflected in `n_modified_contracts` anyway, which is deviation from the correct value calculation. This should be clearly documented.

Reply:
Probably there are some other different cases that this can affect.

Reply:
> In case someone interacts with the token contract at all, it won't be reflected in `n_modified_contracts` anyway, which is deviation from the correct value calculation. This should be clearly documented.

Good catch, this is a bit problematic.

As discussed offline, we won't have to deal with gas-related issues if we put STRK contract directly in cheatnet testing state.